engine/configurator/conf_cyaml.h

```diff
@@ -0,0 +1,806 @@
+    char *value; /* optional */
+} instance_type;
+
+/* Structure for storing backup entry */
+typedef struct backup_entry {
+/* The only one of next pointers is not NULL */
+    object_type *object;
+    instance_type *instance;
+} backup_entry;
+
+/* Structure for storing the whole backup*/
+typedef struct backup_seq {
+    backup_entry *entries;
+    uint entries_count;
+} backup_seq;
+
+/* Structure for storing condition */
+typedef struct cond_entry{ /* cyaml doesn't know about this structure */
+    char *if_cond;
+    struct history_seq *then_cond;
+    struct history_seq *else_cond;
+} cond_entry;
+
+/* Structure for storing entry of history.
+ * The only one pointer in not equal to NULL */
+typedef struct history_entry {
+    char *comment; /* cyaml doesn't know about this field */
+
+    char **incl; /* cyaml doesn't know about this field */
+    uint incl_count;
+
+    cond_entry *cond; /* cyaml doesn't know about this field */
+
+    object_type *reg; /* optional */
+    uint reg_count;
+
+    object_type *unreg; /* optional */
+    uint unreg_count;
+
+    instance_type *add; /* optional */
+    uint add_count;
+
+    instance_type *get; /* optional */
+    uint get_count;
+
+    instance_type *delete; /* optional */
+    uint delete_count;
+
+    instance_type *copy; /* optional */
+    uint copy_count;
+
+    instance_type *set; /* optional */
+    uint set_count;
+
+    char *reboot_ta; /* optional */
+} history_entry;
+
+/* Structure for storing the whole history */
+typedef struct history_seq {
+    history_entry *entries;
+    uint entries_count;
+} history_seq;
+
+/******************************************************************************
+ * CYAML schema to tell libcyaml about both expected YAML and data structure.
+ *
+ * (Our CYAML schema is just a bunch of static const data.)
+ ******************************************************************************/
+
+/* Mapping from instance or object strings to enum values for schema. */
+static const cyaml_strval_t scope_strings[] = {
+        {"object",   CFG_OBJECT},
+        {"instance", CFG_INSTANCE},
+};
+
+/* Mapping from "access" strings to enum values for schema. */
+static const cyaml_strval_t access_strings[] = {
+        {"read_write",  CFG_READ_WRITE },
+        {"read_only",   CFG_READ_ONLY  },
+        {"read_create", CFG_READ_CREATE},
+};
+
+/* Mapping from "type" strings to flag enum for schema. */
+static const cyaml_strval_t type_strings[] = {
+    {"none",    CVT_NONE},
+    {"bool",    CVT_BOOL},
+    {"int8",    CVT_INT8},
+    {"uint8",   CVT_UINT8},
+    {"int16",   CVT_INT16},
+    {"uint16",  CVT_UINT16},
+    {"int32",   CVT_INT32},
+    {"integer", CVT_INT32},
+    {"uint32",  CVT_UINT32},
+    {"int64",   CVT_INT64},
+    {"uint64",  CVT_UINT64},
+    {"string",  CVT_STRING},
+    {"address", CVT_ADDRESS},
+};
+
+/* Mapping from instance or object strings to enum values for schema. */
+static const cyaml_strval_t no_parent_dep_strings[] = {
```

Comment:
Static object normally should not appear in headers. What is the reason?

Reply:
I think it was a mistake. I've removed the definitions to `conf_cyaml.c`.